src/internal/connector/exchange/service_restore.go

```diff
@@ -223,10 +230,19 @@ func RestoreMailMessage(
 	return info, nil
 }
 
-// attachmentBytes is a helper to retrieve the attachment content from a models.Attachmentable
-// TODO: Revisit how we retrieve/persist attachment content during backup so this is not needed
-func attachmentBytes(attachment models.Attachmentable) []byte {
-	return reflect.Indirect(reflect.ValueOf(attachment)).FieldByName("contentBytes").Bytes()
+// GetAttachmentBytes is a helper to retrieve the attachment content from a models.Attachmentable
+func GetAttachmentBytes(attachment models.Attachmentable) ([]byte, error) {
+	bi, err := attachment.GetBackingStore().Get("contentBytes")
+	if err != nil {
+		return nil, err
+	}
+
+	bts, ok := bi.([]byte)
+	if !ok {
+		return nil, clues.New("attachment content is not a byte array")
```

Comment:
when possible, please print out the actual type on typecast failures.  "is not correct type" isn't helpful as an insight, because it doesn't expose the type that we need to handle to fix the situation.
```suggestion
		return nil, clues.New(fmt.Sprintf("unexpected type for attachment content: %T", bi))
```